src/ImageSharp.Drawing/Processing/Text/Processors/DrawTextProcessor.cs

```diff
@@ -0,0 +1,509 @@
+
+                            int end = operation.Map.Height;
+
+                            int maxHeight = source.Height - startY;
+                            end = Math.Min(end, maxHeight);
+
+                            for (int row = fistRow; row < end; row++)
+                            {
+                                int y = startY + row;
+                                Span<float> span = buffer.GetRowSpan(row).Slice(offSetSpan);
+                                app.Apply(span, startX, y);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        private struct DrawingOperation
+        {
+            public IBuffer2D<float> Map { get; set; }
+
+            public Point Location { get; set; }
+        }
+
+        private class CachingGlyphRenderer : IGlyphRenderer, IDisposable
+        {
+            private PathBuilder builder;
+
+            private Point currentRenderPosition = default(Point);
+            private int currentRenderingGlyph = 0;
+            private int offset = 0;
+            private PointF currentPoint = default(PointF);
+            private HashSet<int> renderedGlyphs = new HashSet<int>();
+            private Dictionary<int, Buffer2D<float>> glyphMap;
+            private Dictionary<int, Buffer2D<float>> glyphMapPen;
+            private bool renderOutline = false;
+            private bool renderFill = false;
+            private bool raterizationRequired = false;
+
+            public CachingGlyphRenderer(MemoryAllocator memoryManager, int size, IPen pen, bool renderFill)
+            {
+                this.MemoryManager = memoryManager;
+                this.Pen = pen;
+                this.renderFill = renderFill;
+                this.renderOutline = pen != null;
+                this.offset = 2;
+                if (this.renderFill)
+                {
+                    this.FillOperations = new List<DrawingOperation>(size);
+                    this.glyphMap = new Dictionary<int, Buffer2D<float>>();
+                }
+
+                if (this.renderOutline)
+                {
+                    this.offset = (int)MathF.Ceiling((pen.StrokeWidth * 2) + 2);
+                    this.OutlineOperations = new List<DrawingOperation>(size);
+                    this.glyphMapPen = new Dictionary<int, Buffer2D<float>>();
+                }
+
+                this.builder = new PathBuilder();
+            }
+
+            public List<DrawingOperation> FillOperations { get; }
+
+            public List<DrawingOperation> OutlineOperations { get; }
+
+            public MemoryAllocator MemoryManager { get; internal set; }
+
+            public IPen Pen { get; internal set; }
+
+            public GraphicsOptions Options { get; internal set; }
+
+            public void BeginFigure()
+            {
+                this.builder.StartFigure();
+            }
+
+            public bool BeginGlyph(RectangleF bounds, int cacheKey)
```

Comment:
Don't we need a dedicated `GlyphParameters` struct/class with proper equals + gethashcode implementations  instead of an integer key here? Collisions may lead to bugs!

Reply:
your right... i'll add it to Fonts get it wired up.